### PR TITLE
Script error inside DomUtils.isDocument (close #467)

### LINKED ISSUE #467: Script error inside DomUtils.isDocument
Reproduced for http://www.reddit.com/ site.

### PR DESCRIPTION
**Pull request by churkin**
\cc @VasilyStrelyaev @AlexanderMoskovkin @miherlosev

### CODE REVIEW

File: src/client/sandbox/code-instrumentation/location/index.js [hunk 1]
```diff
@@ -11,7 +11,14 @@ export default class LocationAccessorsInstrumentation extends SandboxBase {
     }
 
     static getLocationWrapper (owner) {
-        return owner[LOCATION_WRAPPER];
+        // NOTE: When owner is cross domain, we can not get his location wrapper, so return the original
+        // location, which will not have access, but it will behave like a real.
```
Comment (VasilyStrelyaev) [inline src/client/sandbox/code-instrumentation/location/index.js]:
```
        // NOTE: When the owner is cross-domain, we cannot get its location wrapper, so we return the original
        // location, which cannot be accessed but behaves like a real one.
```

File: test/client/fixtures/sandbox/code-instrumentation/location-test.js [hunk 2]
```diff
@@ -170,3 +170,22 @@ test('create location wrapper before iframe loading', function () {
 
     iframe.parentNode.removeChild(iframe);
 });
+
+module('regression');
+
+asyncTest('getting location of a cross domain window (GH-467)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/sandbox/code-instrumentation/location-test.js]:
```
asyncTest('getting location of a cross-domain window (GH-467)', function () {
```

File: test/client/fixtures/utils/dom-test.js [hunk 3]
```diff
@@ -404,6 +404,23 @@ test('isHammerheadAttr', function () {
 
 module('regression');
 
+asyncTest('isDocument for cross domain object (GH-467)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/utils/dom-test.js]:
```
asyncTest('isDocument for cross-domain object (GH-467)', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by churkin** [comment]
FPR

**Comment by churkin** [comment]
\сс @kirovboris

File: test/client/fixtures/sandbox/code-instrumentation/location-test.js [hunk 4]
```diff
@@ -170,3 +170,36 @@ test('create location wrapper before iframe loading', function () {
 
     iframe.parentNode.removeChild(iframe);
 });
+
+module('regression');
+
+if (!browserUtils.isFirefox) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/code-instrumentation/location-test.js]: Why do we need this condition?

**Comment by kirovboris** [comment]
lgtm

**Comment by churkin** [comment]
FPR

**Pull request merged**